compiler/aarch64/codegen/UnaryEvaluator.cpp

```diff
@@ -210,3 +210,51 @@ TR::Register *OMR::ARM64::TreeEvaluator::iu2lEvaluator(TR::Node *node, TR::CodeG
    {
    return extendToIntOrLongHelper(node, TR::InstOpCode::ubfmx, 31, cg);
    }
+
+TR::Register *OMR::ARM64::TreeEvaluator::l2aEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Compilation *comp = cg->comp();
+
+   if (comp->useCompressedPointers())
+      {
+      // pattern match the sequence under the l2a
+      //    iaload f      l2a                       <- node
+      //       aload O       ladd
+      //                       lshl
+      //                          i2l
+      //                            iiload f        <- load
+      //                               aload O
+      //                          iconst shftKonst
+      //                       lconst HB
+      //
+      // -or- if the load is known to be null
+      //  l2a
+      //    i2l
+      //      iiload f
+      //         aload O
+      //
+      TR::Node *firstChild = node->getFirstChild();
+      bool hasCompPtrs = false;
+      if (firstChild->getOpCode().isAdd() &&
+          firstChild->containsCompressionSequence())
+         hasCompPtrs = true;
+      else if (node->isNull())
+         hasCompPtrs = true;
+
+      // after remat changes, this assume is no longer valid
+      //
+      ///TR_ASSERT(hasCompPtrs, "no compression sequence found under l2a node [%p]\n", node);
```

Comment:
If the assert is no longer valid, can you remove all the code that sets `hasCompPtrs` ?

Reply:
Removed the whole comment.